src/_pytest/warnings.py

```diff
@@ -113,6 +114,9 @@ def catch_warnings_for_item(config, ihook, when, item):
             ihook.pytest_warning_captured.call_historic(
                 kwargs=dict(warning_message=warning_message, when=when, item=item)
             )
+            ihook.pytest_warning_recorded.call_historic(
+                kwargs=dict(warning_message=warning_message, nodeid=nodeid, when=when)
```

Comment:
Looks like a bug, I commented on https://github.com/pytest-dev/pytest/issues/4049. Thank you for pointing this out @bluetech

Reply:
Looks like the original implementation meant to pass `None` ( as I can't reliably get the stacklevel in all cases ), and assumed that it would be assumed. Will change so that it explicitly sets to `None`